ONSdigital/ras-integration-tests
# Us002 load and view collection exercises

PR blurb courtesy of #4

Added test for https://trello.com/c/TZoUzqkn/24-us002b-view-collection-exercises-13pts

To test
1. Checkout response-operations-ui
2. Hack `surveys.py` to look like the example below (note: this will mean that scenario `Ensure collection exercise exists for a survey` will always pass due to this)
3. Install and run response-operations-ui with `pipenv install && pipenv run python run.py`
4. Run tests with be pipenv run behave
```
@app.route('/surveys/<short_name>', methods=['GET'])
def view_survey(short_name):
    survey_details = {
  "collection_exercises": [
        {
          "id": "14fb3e68-4dca-46db-bf49-04b84e07e77c",
          "name": "QBS_1803",
          "scheduledExecutionDateTime": "2017-09-10T23:00:00.000+0000",
          "exerciseRef": "1803",
          "userDescription": "9 March 2018"
        },
        {
          "id": "14fb3e68-4dca-46db-bf49-04b84e07e77c",
          "name": "QBS_1806",
          "scheduledExecutionDateTime": "2017-09-10T23:00:00.000+0000",
          "exerciseRef": "1806",
          "userDescription": "15 June 2018"
        },
        {
          "id": "14fb3e68-4dca-46db-bf49-04b84e07e77c",
          "name": "QBS_1809",
          "scheduledExecutionDateTime": "2017-09-10T23:00:00.000+0000",
          "exerciseRef": "1809",
          "userDescription": "14 September 2018"
        },
        {
          "id": "14fb3e68-4dca-46db-bf49-04b84e07e77c",
          "name": "QBS_1812",
          "scheduledExecutionDateTime": "2017-09-10T23:00:00.000+0000",
          "exerciseRef": "1812",
          "userDescription": "14 December 2018"
        }
  ],
  "survey": {
    "id": "cb0711c3-0ac8-41d3-ae0e-567e5ea1ef87",
    "longName": "Quarterly Business Survey",
    "shortName": "QBS",
    "surveyRef": "139",
    "legalBasis": "Statistics of Trade Act 1947"
  }
}

    return render_template('survey.html',
                           survey=survey_details['survey'],
                           collection_exercises=survey_details['collection_exercises'])
```
